src/AzurePipelines/PipelineCachingCacheClient.cs

```diff
@@ -226,7 +226,7 @@ internal sealed class PipelineCachingCacheClient : CacheClient
                 foreach (IGrouping<FileRealizationMode, ContentHashWithPath>? tempFilesByRealizationMode in tempFiles.GroupBy(f => GetFileRealizationMode(f.Path.Path)))
                 {
                     FileRealizationMode realizationMode = tempFilesByRealizationMode.Key;
-                    FileAccessMode accessMode = realizationMode == FileRealizationMode.Copy
```

Comment:
Hmm on the "into cache" path we will want to verify hashes externally but right note is rather have CoW

Reply:
Doesn't "Put" always verify the hash?